docs/docs/tutorial/creating-a-source-plugin/part-2/index.mdx

```diff
@@ -0,0 +1,682 @@
+    }
+  } satisfies NodeInput
+
+  gatsbyApi.actions.createNode(node)
+}
+```
+
+<Collapsible
+  summary={<em>Explanation on each required node field</em>}
+>
+
+Here's a short explanation on each required field for your convenience:
+
+- `id`: The node's ID that Gatsby uses to track it. It must be a globally unique ID. You should use the [`createNodeId`](/docs/reference/config-files/node-api-helpers/#createNodeId) helper to create it.
+- `internal`: An object that only allows a defined set of keys (see API reference)
+  - `type`: The name of the node type
+  - `contentDigest`: A hash (or short digital summary) of the contents of a node. The digest should be unique to the node since it's used for caching. If the node changes, this digest should also change. You can use the [`createContentDigest`](/docs/reference/config-files/node-api-helpers/#createContentDigest) helper to create that hash.
+  
+    If your node contains information like a revision ID or an `lastUpdated` timestamp, you should use this instead of creating your own content digest. This will lead to lower CPU utilization and still achieve the goal of invalidating the cache for the node when said node changes.
+
+You can use the [`createNode` API reference](/docs/reference/config-files/actions/#createNode) in the future if you're creating more advanced node types, e.g. ones where you have to define the [`mediaType`](/docs/tutorial/creating-a-source-plugin/part-7/#mime-types).
+
+</Collapsible>
+
+Afterwards one is able to query the `TypeName` type with a `hello` key:
+
+```graphql
+query {
+  typeName {
+    hello
+  }
+}
+```
+
+### Task: Define constants
+
+Since you'll need to use the name of your node types in a couple of places, it's easy to introduce typos while using them. Create a `constants.ts` file to define your names once and reuse them everywhere. This way you can also more easily change the names after the fact. Neat!
+
+<Announcement>
+
+You don't always need constants for your node types. It depends on your data source and the data you have to work with.
+
+If you're working with a more barebones API you might need to manually map endpoints and/or data to node types. Working with a CMS on the other hand node types will often be dynamic and defined through the CMS data. In those instances you won't need to manually define types as the CMS will drive that.
+
+</Announcement>
+
+Add the following to your new `constants.ts` file:
+
+```ts:title=plugin/src/constants.ts
+export const NODE_TYPES = {
+  Post: `Post`,
+  Author: `Author`,
+} as const
+```
+
+<Announcement>
+
+**TypeScript tip:** You're using a [`const` assertion](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) here to make the object read-only and tell the compiler to infer the most specific type it can. In this specific instance you use `as const` to enable a `enum`-like pattern instead of directly using TypeScript's `enum`.
```

Comment:
Small typo:
```suggestion
**TypeScript tip:** You're using a [`const` assertion](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) here to make the object read-only and tell the compiler to infer the most specific type it can. In this specific instance you use `as const` to enable an `enum`-like pattern instead of directly using TypeScript's `enum`.
```